pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4360,19 +4363,22 @@ static __always_inline int tc_probe(struct __sk_buff *skb, bool ingress) {
     //
     // See bpf_skb_event_output in net/core/filter.c.
     u64 flags = BPF_F_CURRENT_CPU;
-    flags |= (u64)skb->len << 32;
-    if (get_config(CONFIG_OPTIONS) & OPT_DEBUG_NET){
+
+    // If not capture, only send the minimal required data to save the
+    // packet. This will be the timestamp (u64), net event_id (u32),
+    // host_tid (u32), comm (16 bytes), packet len (u32), and ifindex (u32)
+    size_t pkt_size = 40;
+
+    if (event_chosen(NET_PACKET)){
+        pkt_size = sizeof(pkt);
         pkt.src_port = __bpf_ntohs(pkt.src_port);
         pkt.dst_port = __bpf_ntohs(pkt.dst_port);
-        bpf_perf_event_output(skb, &net_events, flags, &pkt, sizeof(pkt));
-    }
-    else {
-        // If not debugging, only send the minimal required data to save the
-        // packet. This will be the timestamp (u64), net event_id (u32),
-        // host_tid (u32), comm (16 bytes), packet len (u32), and ifindex (u32)
-        bpf_perf_event_output(skb, &net_events, flags, &pkt, 40);
     }
 
+    if ((get_config(CONFIG_OPTIONS) & OPT_CAPTURE_PACKETS)){
```

Comment:
Doing this is not enough I'm afraid.
In the case of multiple interfaces receieved as an input, how do you know if this specific interface was chosen for capture or not?